### PR TITLE
Cut WeePack out of core.

### PR DESCRIPTION
**Pull request by htmldoug**
This inverts the dependency graph for msgpack. WeePickle no longer depends on WeePack. WeePack now depends on WeePickle.

There are 3 ASTs in this library whose members all have identical names:
- `com.rallyhealth.weejson.v1.Obj`
- `com.rallyhealth.weepack.v1.Obj`
- `com.rallyhealth.weejson.v1.BufferedValue.Obj`

`weejson` and `weepack` are so similar that importing the right one a pain.

MsgPack seems like a fine format, but we don't use it today. It parses faster than json, but generates slightly slower. This should probably be an optional dependency.

### CODE REVIEW

File: bench/src-jvm/bench/bench.scala [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+/**
+  * mill bench.jvm.runJmh GeneratorBench
+  *
+  * {{{
+  * Benchmark                       Mode  Cnt    Score    Error  Units
+  * GeneratorBench.jsonBytes       thrpt    9  273.819 ± 22.893  ops/s
+  * GeneratorBench.jsonString      thrpt    9  255.608 ± 11.505  ops/s
+  * GeneratorBench.msgpackJackson  thrpt    9  144.934 ±  2.270  ops/s
+  * GeneratorBench.msgpackScala    thrpt    9  219.204 ±  4.257  ops/s
```
Comment (htmldoug) [inline bench/src-jvm/bench/bench.scala]: My best guess is that it's missing a fast-path for ASCII that jackson-smile happily exploits.

File: bench/src-jvm/bench/bench.scala [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  * GeneratorBench.smile           thrpt    9  352.632 ± 14.197  ops/s
+  * }}}
+  *
+  * java 11:
+  * {{{
+  * Benchmark                       Mode  Cnt    Score    Error  Units
+  * GeneratorBench.jsonBytes       thrpt   15  238.335 ± 11.777  ops/s
+  * GeneratorBench.jsonString      thrpt   15  240.125 ±  7.871  ops/s
+  * GeneratorBench.msgpackJackson  thrpt   15  181.195 ±  5.774  ops/s
+  * GeneratorBench.msgpackScala    thrpt   15  304.540 ±  2.225  ops/s
```
Comment (htmldoug) [inline bench/src-jvm/bench/bench.scala]: wow, I guess the new java ASCII encoding really picks up some slack here.

File: bench/src-jvm/bench/bench.scala [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  * ParserBench.smile           thrpt    9  309.749 ± 5.861  ops/s
+  * }}}
+  *
+  * java 11:
+  * {{{
+  * Benchmark                    Mode  Cnt    Score    Error  Units
+  * ParserBench.jsonBytes       thrpt   15  245.665 ±  3.202  ops/s
+  * ParserBench.jsonString      thrpt   15  213.312 ±  5.250  ops/s
+  * ParserBench.msgpackJackson  thrpt   15  205.738 ±  2.789  ops/s
+  * ParserBench.msgpackScala    thrpt   15  422.313 ± 17.172  ops/s
```
Comment (htmldoug) [inline bench/src-jvm/bench/bench.scala]: wow, I guess I owe this parser an apology.

Comment (russellremple) [reply bench/src-jvm/bench/bench.scala]: Boo Jackson

File: build.sc [hunk 4]
```diff
@@ -339,6 +339,8 @@ object bench extends Module {
       ivy"com.fasterxml.jackson.module::jackson-module-scala:2.9.10",
       ivy"com.fasterxml.jackson.core:jackson-databind:2.9.4",
       ivy"com.lihaoyi::upickle:0.9.8",
+      ivy"org.msgpack:jackson-dataformat-msgpack:0.8.20",
+      ivy"com.fasterxml.jackson.dataformat:jackson-dataformat-smile:2.10.2",
     )
   }
 }
```
Comment (russellremple) [inline build.sc]: Still planning to move to SBT someday?

File: core/src/com/rallyhealth/weepickle/v1/core/JsVisitor.scala [hunk 5]
```diff
@@ -58,6 +57,6 @@ trait JsVisitor[-T, +J] extends Visitor[T, J] {
   def visitChar(c: Char): J = visitString(c.toString)
 
   def visitTimestamp(instant: Instant): J = {
-    visitString(DateTimeFormatter.ISO_INSTANT.format(instant))
+    visitString(instant.toString)
```
Comment (russellremple) [inline core/src/com/rallyhealth/weepickle/v1/core/JsVisitor.scala]: +1

File: build.sc (hunk 4 above)
Comment (htmldoug) [reply build.sc]: Depends how much mill fights publishing to a non-maven-central repo. I've been doing __.publishLocal and uploading a zip to artifactory using the web GUI like a savage.

File: bench/src-jvm/bench/bench.scala (hunk 3 above)
Comment (htmldoug) [reply bench/src-jvm/bench/bench.scala]: Yeah, I expected quite a bit more here. I guess the stuff at the edges of the jackson ecosystem get optimized at varying degrees.

**Pull request merged**